transfer_test.go

```diff
@@ -0,0 +1,161 @@
+		err = configCli.WithoutCredentials().Exec("rm", "target", "--quiet")
+		assert.NoError(t, err)
+	}
+	*tests.JfrogUrl = utils.AddTrailingSlashIfNeeded(*tests.JfrogUrl)
+	err = tests.NewJfrogCli(execMain, "jfrog config", "--access-token="+*tests.JfrogTargetAccessToken).Exec("add", "target", "--interactive=false", "--artifactory-url="+*tests.JfrogTargetUrl+tests.ArtifactoryEndpoint)
+	assert.NoError(t, err)
+
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("curl", "-XPOST", "/api/plugins/reload"))
+	return func() {
+		cleanArtifactory()
+		inttestutils.CleanTargetRepos(targetArtifactoryCli)
+		clientTestUtils.SetEnvAndAssert(t, coreutils.HomeDir, oldHomeDir)
+		clientTestUtils.RemoveAllAndAssert(t, newHomeDir)
+		tests.CleanFileSystem()
+	}
+}
+
+func TestTransferTwoRepos(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	// Upload files
+	runRt(t, "upload", "testdata/a/*", tests.RtRepo1)
+	runRt(t, "upload", "testdata/a/b/*", tests.RtRepo2)
+
+	// Verify files were uploaded to the source Artifactory
+	repo1Spec, err := tests.CreateSpec(tests.SearchAllRepo1)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, t)
+	repo2Spec, err := tests.CreateSpec(tests.SearchRepo2)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, t)
+
+	// Execute transfer-files
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.RtRepo1+";"+tests.RtRepo2))
+
+	// Verify files were transferred to the target Artifactory
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, targetServerDetails, t)
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, targetServerDetails, t)
+}
+
+func TestTransferExcludeRepo(t *testing.T) {
+	cleanUp := initTransferTest(t)
+	defer cleanUp()
+
+	// Upload files
+	runRt(t, "upload", "testdata/a/*", tests.RtRepo1)
+	runRt(t, "upload", "testdata/a/b/*", tests.RtRepo2)
+
+	// Verify files were uploaded to the source Artifactory
+	repo1Spec, err := tests.CreateSpec(tests.SearchAllRepo1)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, t)
+	repo2Spec, err := tests.CreateSpec(tests.SearchRepo2)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetTransferExpectedRepo2(), repo2Spec, t)
+
+	// Execute transfer-files
+	assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.RtRepo1+";"+tests.RtRepo2, "--exclude-repos="+tests.RtRepo2))
+
+	// Verify repo1 files were transferred to the target Artifactory
+	inttestutils.VerifyExistInTargetArtifactory(tests.GetTransferExpectedRepo1(), repo1Spec, targetServerDetails, t)
+	inttestutils.VerifyExistInTargetArtifactory([]string{}, repo2Spec, targetServerDetails, t)
+}
+
+func TestTransferMaven(t *testing.T) {
```

Comment:
An improvement to this test would be to assert the delayed files do not exists in target until handled. Not sure if this is easily feasible.

Reply:
I don't think it is possible, because Artifactory can index the Maven repository at the second we upload the pom.xml.

Reply:
I wasn't referring to the metadata files, but to the delayed files (pom.xml itself)